services/traction/api/db/models/v1/holder.py

```diff
@@ -286,3 +286,214 @@ async def get_by_revocation_ids(
+
+        Returns: The Traction HolderPresentation (db) record
+
+        Raises:
+          NotFoundError: if the HolderPresentation cannot be found by ID
+        """
+
+        q = (
+            select(cls)
+            .where(cls.tenant_id == tenant_id)
+            .where(cls.presentation_exchange_id == presentation_exchange_id)
+            .options(selectinload(cls.contact))
+        )
+        q_result = await db.execute(q)
+        db_rec = q_result.scalar_one_or_none()
+        if not db_rec:
+            raise NotFoundError(
+                code="holder_presentation.presentation_exchange_id_not_found",
+                title="Holder Presentation does not exist",
+                detail=f"Holder Presentation does not exist for presentation exchange id<{presentation_exchange_id}>",  # noqa: E501
+            )
+        return db_rec
+
+    @classmethod
+    async def list_by_contact_id(
+        cls: "HolderPresentation",
+        db: AsyncSession,
+        tenant_id: uuid.UUID,
+        contact_id: uuid.UUID,
+    ) -> List["HolderPresentation"]:
+        """List by Contact ID.
+
+        Find and return list of Holder Presentation records for Contact.
+
+          tenant_id: Traction ID of tenant making the call
+          contact_id: Traction ID of Contact
+
+        Returns: List of Traction HolderPresentation (db) records in descending order
+        """
+
+        q = (
+            select(cls)
+            .where(cls.contact_id == contact_id)
+            .where(cls.tenant_id == tenant_id)
+            .options(selectinload(cls.contact))
+            .order_by(desc(cls.updated_at))
+        )
+        q_result = await db.execute(q)
+        db_recs = q_result.scalars().all()
+        return db_recs
+
+    @classmethod
+    async def list_by_tenant_id(
+        cls: "HolderPresentation",
+        db: AsyncSession,
+        tenant_id: uuid.UUID,
+    ) -> List["HolderPresentation"]:
+        """List by Tenant ID.
+
+        Find and return list of Holder Presentation records for Tenant.
+
+          tenant_id: Traction ID of tenant making the call
+
+        Returns: List of Traction Holder Presentation (db) records in descending order
+        """
+
+        q = (
+            select(cls)
+            .where(cls.tenant_id == tenant_id)
+            .options(selectinload(cls.contact))
+            .order_by(desc(cls.updated_at))
+        )
+        q_result = await db.execute(q)
+        db_recs = q_result.scalars().all()
+        return db_recs
+
+    @classmethod
+    async def list_by_thread_id(
+        cls: "HolderPresentation",
+        db: AsyncSession,
+        tenant_id: uuid.UUID,
+        thread_id: str,
+    ) -> List["HolderPresentation"]:
+        """List by Thread ID.
+
+        Find and return list of Holder Presentation records for Thread ID.
+
+          tenant_id: Traction ID of tenant making the call
+          thread_id: AcaPy Thread ID of Issuer Presentation
```

Comment:
Is this supposed to read "Holder Presentation"? Or is Issuer right for this one...

Reply:
nope, copy/paste...

Reply:
fixed.